src/main/java/uk/gov/pay/connector/queue/tasks/handlers/DeleteStoredPaymentDetailsTaskHandler.java

```
@@ -29,9 +30,13 @@ public void process(String agreementExternalId, String paymentInstrumentExternal
         var agreement = agreementService.findByExternalId(agreementExternalId);
         var paymentInstrument = paymentInstrumentService.findByExternalId(paymentInstrumentExternalId);
         PaymentProvider paymentProvider = providers.byName(PaymentGatewayName.valueFrom(agreement.getGatewayAccount().getGatewayName()));
-
         DeleteStoredPaymentDetailsGatewayRequest request = new DeleteStoredPaymentDetailsGatewayRequest(agreement, paymentInstrument);
-        var response = paymentProvider.deleteStoredPaymentDetails(request);
-        LOGGER.info("Processed deleteStoredPaymentDetails task with response: {}", response.toString());
+        try {
+            var response = paymentProvider.deleteStoredPaymentDetails(request);
+            LOGGER.info("Processed deleteStoredPaymentDetails task with response: {}", response.toString());
+        } catch (GatewayException ex) {
```

I don't think we should catch this exception as this will mean we silently fail to delete the token and don't retry. We should propagate it up to the TaskQueueMessage handler, where it will be caught and handled appropriately.
If `TaskQueueMessageHandler` handles an exception, it will not mark the SQS message as processed, and so it will be retried some time later. If processing the message repeatedly fails, the message will be sent to a dead letter queue and we will get alerted to Zendesk for support to investigate.

---

changed as requested.